src/FakeItEasy/Configuration/AnyCallCallRule.cs

```diff
@@ -37,7 +39,9 @@ protected override bool OnIsApplicableTo(IFakeObjectCall fakeObjectCall)
         {
             return
                 this.argumentsPredicate(fakeObjectCall.Arguments) &&
-                    (this.ApplicableToMembersWithReturnType == null || this.ApplicableToMembersWithReturnType.Equals(fakeObjectCall.Method.ReturnType));
+                    (this.ApplicableToMembersWithReturnType == null 
+                    || this.ApplicableToMembersWithReturnType.Equals(fakeObjectCall.Method.ReturnType)
+                    || this.ApplicableToAllNonVoidReturnTypes);

```

Comment:
Should check that the return type is not `void` when `ApplicableToAllNonVoidReturnTypes` is true